container/lxd/manager.go

```diff
@@ -336,3 +344,44 @@ func (m *containerManager) ReplaceOrAddInstanceProfile(instId, oldProfile, newPr
 	}
 	return m.LXDProfileNames(instId)
 }
+
+// AssignProfiles implements environs.LXDProfiler.
+func (m *containerManager) AssignProfiles(instId string, profilesNames []string, profilePosts []lxdprofile.ProfilePost) (current []string, err error) {
+	defer func() ([]string, error) {
+		if err != nil {
```

Comment:
I'm not entirely sure this works as expected, as all `err` below this use `:=`, so nothing will get past here.

I wonder if we can do this a better way and wrap all exits you want to pass this back in a function.

```go
report := func(err error) ([]string, error) {
    current, err1 := m.LXDProfileNames(instId)
    if err1 != nil {
          logger.Errorf("secondary error, retrieving profile names: %s", err1)
    }
    return current, err
}
...
return report(err)
```

Reply:
done, this works better, ty for the idea.